lib/https.js

```diff
@@ -100,6 +126,28 @@ Agent.prototype.getName = function(options) {
   return name;
 };
 
+Agent.prototype._getSession = function _getSession(key) {
+  return this._sessionCache.map[key];
+};
+
+Agent.prototype._cacheSession = function _cacheSession(key, session) {
+  // Fast case - update existing entry
+  if (this._sessionCache.map[key]) {
+    this._sessionCache[key] = session;
+    return;
+  }
+
+  // Put new entry
+  if (this._sessionCache.list.length >= this.maxCachedSessions) {
+    const oldKey = this._sessionCache.list.shift();
```

Comment:
Premature optimization? This is a list of length 100, and it is unlikely that it will ever become a bottleneck. Especially, considering that how much time TLS handshake takes.

cc @bnoordhuis @trevnorris